internal/git/gitsync_processor.go

```diff
@@ -467,6 +595,34 @@ func NewGitSyncProcessor(ctx context.Context, gitSync *v1alpha1.GitSync, kubeCli
 	return processor, nil
 }
 
+// getLocalRepoPath will return the local path where repo will be cloned, if not set the use /tmp as default dir.
+func getLocalRepoPath(gitSyncName, repoName string) string {
+	localRepoPath := os.Getenv("LOCAL_REPO_PATH")
+	if localRepoPath != "" {
+		return fmt.Sprintf("%s/%s-%s", localRepoPath, gitSyncName, repoName)
+	} else {
+		return fmt.Sprintf("/tmp/%s-%s", gitSyncName, repoName)
+	}
+}
+
+// getRepositoryName parse the url and return the repo name from it.
+func getRepositoryName(url string) (string, error) {
+	// Remove protocol and hostname
```

Comment:
sorry, I think I didn't clarify what I meant when I said that we should use the GitSync name to RepositoryPath name. By "Repository Path name", I was simply referring to this field: https://github.com/numaproj-labs/numaplane/blob/main/api/v1alpha1/gitsync_types.go#L73

Reply:
okay, Sorry I meant it wrong!! I'll make this change accordingly.

Reply:
But we need to make the name field as mandatory right? I don't think currently it is.

Reply:
well, there's [validation](https://github.com/numaproj-labs/numaplane/blob/main/internal/controller/gitsync_controller.go#L292) for it

Reply:
ohh, Yaa remember now that shubham added it, I was checking `gitsync_processor_test.go` tests and seen that many tests are missing `name` so thought validation is not there, but validation is in reconsile method

Reply:
I guess one thing I should mention is that @xdevxy has a PR open right now which will remove the multiple RepositoryPath concept from a GitSync, so really we won’t end up needing to add the RepositoryPath name at all and the GitSync name will be sufficient for uniqueness

Reply:
Sure, I have update the path generation method